sources/web/datalab/polymer/test/SettingsManager-test.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+declare function assert(condition: boolean): null;
+declare function assert(condition: boolean, message: string): null;
+
+/// <reference path="../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../node_modules/@types/chai/index.d.ts" />
+
+describe('SettingsManager', () => {
+
+  describe('loadConfigToWindowDatalab', () => {
+    const initialWindowDatalab = window.datalab;
+    beforeEach(() => {
+      window.datalab = initialWindowDatalab;  // Reset to undefined
+    });
+
+    it('resolves immediately if window.datalab is already set', async () => {
+      window.datalab = { foo: 'bar' };
+      SettingsManager.getAppSettingsAsync = (forceRefresh?: boolean) => {
```

Comment:
I'll give it a try. It compiles on my local machine, but my local machine hasn't been complaining about a lot of stuff that then fails on Travis.